src/services/ibkr_select_contracts.py

```
@@ -0,0 +1,599 @@
+    def build_spec(self, request: ContractSelectionRequest) -> Contract:
+        raise NotImplementedError(
+            "FutureOptionContractSelector uses select() with IB-dependent chain lookup."
+        )
+
+    def select(self, ib: IB, request: ContractSelectionRequest) -> tuple[Contract, int]:
+        spec = self._build_fop_spec(ib, request)
+        _qualify_contracts(ib, spec)
+        if spec.conId is None:
+            raise RuntimeError(
+                f"No FOP contract found for {_build_lookup_context(request)}."
+            )
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)